ocs_ci/ocs/resources/clientprofile.py

```diff
@@ -0,0 +1,118 @@
+import logging
+import tempfile
+
+from ocs_ci.ocs import constants, ocp
+from ocs_ci.framework import config
+from ocs_ci.utility import templating
+from ocs_ci.utility.templating import dump_data_to_temp_yaml
+
+logger = logging.getLogger(__name__)
+
+
+class ClientProfile:
+    """
+    Base ClientProfile class
+    """
+
+    def __init__(self, client_profile_name, consumer_context=None):
+
+        self.consumer_context = consumer_context
+        self.name = client_profile_name
+        self.ocp = ocp.OCP(
+            resource_name=self.name,
+            kind=constants.CLIENT_PROFILE,
+            namespace=config.cluster_ctx.ENV_DATA["cluster_namespace"],
+        )
+        if self.consumer_context:
+            self.provider_context = config.cluster_ctx.MULTICLUSTER[
+                "multicluster_index"
+            ]
+        else:
+            self.provider_context = None
+
+    def get_ceph_connection_reference(self):
+        """
+        Get the CephConnectionReference name
+
+        Returns:
+            str: CephConnectionReference name
+        """
+        with config.RunWithConfigContext(self.consumer_context):
```

Comment:
What would happen, if the `self.consumer_context` will be `None`? (Not sure, if that is possible, but assuming that yes, since the argument in `__init__` is optional, default to `None`.)
The same apply for the following occurrences too.